evennia/contrib/aisystem/README.md

```diff
@@ -0,0 +1,661 @@
+                 during its last tick.
+
+running\_child (int) - the currently running child of a MemSequence or
+                      MemSelector node.
+
+ticks (int) - for repeater nodes, this is the number of repeats that have
+              already been performed since the repeater started running. For
+              limiter nodes, this is the number of repeats left before the
+              limiter shuts down.
+
+watchers (list) - a list of all the players that are currently watching the node
+
+Each blackboard also contains the following global data:
+
+resources (dict) - a dictionary with the names of resources as keys and the
+                   things using them as values. See the resource allocator
+                   node for details about this.
+
+errors (dict) - a dictionary with the hashes of nodes as keys and the error
+                strings they returned during the most recent iteration of the
+                blackboard as values.
+
+### Setting up the ai system
+
+First, subclass the classes of any AI-using objects, characters or other
+entities from the AIObject class:
+
+```
+from evennia.contrib.aisystem import AIObject
+
+class MyClass(AIObject, <other superclasses>)
+```
+
+Also set up any script classes that you want to have an AI blackboard by
+subclassing them from the AIScript class, or simply use the AIScript class
+directly.
+
+```
+from evennia.contrib.aisystem import AIScript
+
+class MyScript(AIScript, <other superclasses>)
+```
+
+You can assign each class that inherits from AIObject or AIScript a property
+called `aitree`, whose value can be either a BehaviorTree or a string that
+references the name or database id of a given tree. For instance:
+
+```
+from evennia.contrib.aisystem import AIScript
+
+class MyScript(AIScript, <other superclasses>)
+    aitree = 'fighter tree'
+```
+
+Also set up the player class of any players that you want to have access to
+the AI system commands by subclassing it from AIPlayer.
+
+```
+from evennia.contrib.aisystem import AIPlayer
+
+new_nodes = [...]
+
+class MyPlayerClass(AIPlayer, <other superclasses>)
+    [...]
+    def at_player_creation(self):
+        super(MyPlayerClass, self). at_player_creation()
+        self.ainodes.update(new_nodes)
+```
+
+Here, new\_nodes is a dictionary whose keys are the names of node classes that
+you will reference in-game when adding new nodes to your trees, and whose items
+are the respective node classes. For instance, if new\_nodes is
+`{'my_node': MyNode}, then you can add a node of class MyNode to the currently
+browsed node by using the "@aiadd my_node 'my node'" command. Node names should
+only contain alphanumeric characters and underscores, never spaces.
+
+Even if they are subclassed from this class, players will only have access to
+the AI system commands if they are wizards, immortals or the superuser.
+
+Finally, you need to load the AI system's command set into your game. One way
```

Comment:
From looking at the commands, it appears the commands are all builder-related commands. It should be pointed out that these are not needed unless for entities that should be able to modify AI states in-game.